src/sentry/static/sentry/app/views/organizationIncidents/details/activity.jsx

```diff
@@ -291,6 +256,53 @@ class ActivityContainer extends React.Component {
     }
   };
 
+  getIndexAndActivityFromState = activity => {
+    // `index` should probably be found, if not let error hit Sentry
+    const index = this.state.activities.findIndex(({id}) => id === activity.id);
+    return [index, this.state.activities[index]];
+  };
+
+  handleDeleteNote = async activity => {
+    const {api, params} = this.props;
+    const {incidentId, orgId} = params;
+
+    const [index, oldActivity] = this.getIndexAndActivityFromState(activity);
+
+    this.setState(state => ({
+      activities: removeFromArrayIndex(state.activities, index),
+    }));
+
+    try {
+      await deleteIncidentNote(api, orgId, incidentId, activity.id);
+    } catch (error) {
+      this.setState(state => ({
+        activities: replaceAtArrayIndex(state.activities, index, oldActivity),
+      }));
+    }
+  };
+
+  handleUpdateNote = async (note, activity) => {
+    const {api, params} = this.props;
+    const {incidentId, orgId} = params;
+
+    const [index, oldActivity] = this.getIndexAndActivityFromState(activity);
+
+    this.setState(state => ({
+      activities: replaceAtArrayIndex(state.activities, index, {
+        ...oldActivity,
```

Comment:
Do you want the new activity here as you're going to send the update off?

Reply:
This extends the oldActivity and updates the comment since that's the only thing that is updated